### DISCUSSION TITLE
Argument #1 ($doctrine) must be of type App\Repository\HistoriqueSondeValeurRepository, null given

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [symfony/symfony#46516] [Q&A] [answered]
Hello everyone ,
i want to do tests in my database but i keep getting this error :  **Argument #1 ($doctrine) must be of type App\Repository\HistoriqueSondeValeurRepository, null given** i have a filter methode that extract value from database depending of dates here is the test: 
![image](https://user-images.githubusercontent.com/101296742/171105511-02a76b69-74a7-4d47-9f89-792d3eb45c5c.png)
and the methode : 
![image](https://user-images.githubusercontent.com/101296742/171105563-314ef7d1-b81a-4578-a9d6-526f3c29b0e9.png)
the findByCurrentDate query:
![image](https://user-images.githubusercontent.com/101296742/171105676-fd9f8b51-d0a2-4740-99a0-fca52c413ea7.png)
thank you !

**Comment by stof** [comment]
Well, make sure that `$this->doctrine` is initialized

**Comment by Elmatbaisaad** [comment, marked as answer]
i had to mock the Repository rather than using the right one like that :
![image](https://user-images.githubusercontent.com/101296742/171173270-413cc566-4b43-4333-8b70-f2d4b2346424.png)
